apps/dashboard/src/contract-ui/tabs/shared-components/listing-drawer.tsx

```diff
@@ -132,18 +132,25 @@ export const ListingDrawer: React.FC<NFTDrawerProps> = ({
                   </Text>
                 </GridItem>
 
-                <GridItem colSpan={3}>
-                  <Heading size="label.md">Price</Heading>
-                </GridItem>
-                <GridItem colSpan={9}>
-                  <Text fontFamily="mono" size="body.md">
-                    {
-                      (renderData as DirectListing).currencyValuePerToken
-                        .displayValue
-                    }{" "}
-                    {(renderData as DirectListing).currencyValuePerToken.symbol}
-                  </Text>
-                </GridItem>
+                {(renderData as DirectListing).currencyValuePerToken && (
```

Comment:
why do we need the type casting here? do we not have a clear type for `renderData`? the listing should have a type that we can switch on?

Reply:
the dashboard is using this component for both English Auction and Direct Listing so..

Reply:
makes sense but then should `renderData` not contain some kind of type that clearly makes into one or the other meaning you could simply do `if renderData.type === "xyz"` -> have the correct type inside the if block?

Reply:
It's better that way agreed! On it